src/asImageWidthSrcSet.ts

```diff
@@ -0,0 +1,102 @@
+import { ImageFieldImage } from "@prismicio/types";
+import {
+	buildURL,
+	buildWidthSrcSet,
+	BuildWidthSrcSetParams,
+} from "imgix-url-builder";
+
+import { imageThumbnail as isImageThumbnailFilled } from "./isFilled";
+
+/**
+ * The return type of `asImageWidthSrcSet()`.
+ */
+type AsImageWidthSrcSetReturnType<Field extends ImageFieldImage> =
+	Field extends ImageFieldImage<"empty">
+		? null
+		: {
+				/**
+				 * The Image field's image URL with Imgix URL parameters (if given).
+				 */
+				src: string;
+
+				/**
+				 * A width-based `srcset` attribute value for the Image field's image
+				 * with Imgix URL parameters (if given).
+				 */
+				srcset: string;
+		  };
```

Comment:
Shouldn't this helper also return a suggested value for the `sizes` attribute?

https://developer.mozilla.org/en-US/docs/Learn/HTML/Multimedia_and_embedding/Responsive_images#resolution_switching_different_sizes

Reply:
I might be wrong, but I don't believe we can suggest a `sizes` attribute. `sizes` effectively refers to the size of the rendered image, which we don't know.

By leaving out a `sizes` attribute, the browser should load the closest matching `srcset` entry based on viewport width. This is as good as we can predict without knowing the size of the rendered image.

We only have enough data to provide something like this (assuming `widths` is `[400, 800, 1600]`:

```
(max-width: 600px) 400px,
(max-width: 600px) 800px,
1600px
```

Which is what the `srcset` value already provides.

If I'm misunderstanding `srcset` and `sizes`, please let me know :)

Reply:
OK got it, if it adapts to the window width without the sizes attribute work for me! (Been some time too since I used those directly ^^)